src/app/drive/components/DeleteBackupDialog/DeleteBackupDialog.tsx

```diff
@@ -0,0 +1,105 @@
+import { useState } from 'react';
+import errorService from 'app/core/services/error.service';
+import { uiActions } from 'app/store/slices/ui';
+import { useAppDispatch, useAppSelector } from 'app/store/hooks';
+import { RootState } from 'app/store';
+import { DriveFolderData as DriveWebFolderData, DriveItemData } from '../../types';
+import i18n from 'app/i18n/services/i18n.service';
+import Button from 'app/shared/components/Button/Button';
+import Modal from 'app/shared/components/Modal';
+import { deleteItemsThunk } from '../../../store/slices/storage/storage.thunks/deleteItemsThunk';
+import { deleteBackupDeviceAsFolder } from '../../../drive/services/folder.service';
+import { backupsThunks } from 'app/store/slices/backups';
+import { SdkFactory } from '../../../core/factory/sdk';
+
+export interface BreadcrumbItemData {
+  id: number;
+  label: string;
+  icon: JSX.Element | null;
+  active: boolean;
+  isFirstPath?: boolean;
+  dialog?: boolean;
+  isBackup?: boolean;
+  onClick?: () => void;
+}
+
+interface BreadcrumbsProps {
+  items: BreadcrumbItemData[];
+  goToFolder: (folderId: number) => void;
+}
+
+const DeleteBackupDialog = (props: BreadcrumbsProps): JSX.Element => {
+  const [isLoading, setIsLoading] = useState(false);
+  const dispatch = useAppDispatch();
+  const isOpen = useAppSelector((state: RootState) => state.ui.isDeleteBackupDialogOpen);
+  const currentDevice = useAppSelector((state) => state.backups.currentDevice);
+  const breadcrumbsItems = props.items;
+  const currentBreadcrumb = props.items[props.items.length - 1];
+  const previousBreadcrumb = props.items[props.items.length - 2];
+
+  const onClose = (): void => {
+    dispatch(uiActions.setIsDeleteBackupDialog(false));
+  };
+
+  const onAccept = async (): Promise<void> => {
+    if (breadcrumbsItems.length === 2) {
+      try {
+        setIsLoading(true);
+        if (currentDevice && 'mac' in currentDevice) dispatch(backupsThunks.deleteDeviceThunk(currentDevice));
+        else {
+          await dispatch(deleteItemsThunk([currentDevice as DriveItemData])).unwrap();
+          await deleteBackupDeviceAsFolder(currentDevice as DriveWebFolderData);
+          dispatch(backupsThunks.fetchDevicesThunk());
+        }
+
+        setIsLoading(false);
+        onClose();
+        props.goToFolder(previousBreadcrumb.id);
+      } catch (err: unknown) {
+        const castedError = errorService.castError(err);
+
+        setIsLoading(false);
+
+        console.log(castedError.message);
```

Comment:
Add sentry error report. Could use `reportError` function that Robert has added to the `error.service.ts` service.

Reply:
done